app/controllers/canvas_controller.rb

```diff
@@ -253,25 +253,25 @@ def update_params
   end
 
   def load_vars
-    @project = Project.find_by_id(params[:id])
-    unless @project
+    @experiment = Experiment.find_by_id(params[:id])
+    unless @experiment
       respond_to do |format|
         format.html { render_404 and return }
         format.any(:xml, :json, :js) { render(json: { redirect_url: not_found_url }, status: :not_found) and return }
       end
     end
 
-    @my_modules = @project.active_modules
+    @my_modules = @experiment.active_modules
   end
 
   def check_edit_canvas
-    unless can_edit_canvas(@project)
+    unless can_edit_canvas(@experiment.project)
```

Comment:
I'd still refactor this permission, to take experiment as a parameter instead of project. Just for the sake of keeping to the current way permissions are done.